index.js

```diff
@@ -0,0 +1,136 @@
+// select elements
+const form = document.querySelector('form');
+const clearAll = document.querySelector('.complete');
+const input = document.getElementById('text');
+const todolistContainer = document.querySelector('.todo-list');
+
+let todoArr = JSON.parse(localStorage.getItem('TODO')) || [];
+let editTodoId = -1;
+
+// add to do
+const addTodo = () => {
+  const inputValue = input.value;
+
+  // check if todo is empty
+  const isempty = inputValue === '';
+  if (isempty) {
+    return;
+  }
+  if (editTodoId >= 0) {
+    todoArr = todoArr.map((todo, index) => ({
+      ...todo,
+      description: index === editTodoId ? inputValue : todo.description,
+    }));
+  } else {
+    todoArr.push({
+      description: inputValue,
+      complete: false,
+      index: todoArr.length + 1,
+    });
+  }
+  input.value = '';
+};
+
+// render todos function
+const renderTodo = () => {
+  // Clear render before a Re-render
+  todolistContainer.innerHTML = '';
+
+  // Render todos
+  todoArr.forEach((todo, index) => {
+    todolistContainer.innerHTML += `
+  <div class="list" id=${index}>
+  <input type="checkbox" data-action="checkbox">
+    <p class="text" data-action="edit">${todo.description}</p>
+    <i class="fa-regular fa-trash-can trash" data-action="delete" ></i>
+</div>
+  `;
+  });
+};
+
+const editTodo = (todoId) => {
+  input.value = todoArr[todoId].description;
+  editTodoId = todoId;
+};
+
+const deleteTodo = (todoId) => {
+  todoArr = todoArr.filter((todo, index) => index !== todoId);
+  editTodoId = -1;
+  for (let i = 0; i < todoArr.length; i += 1) {
+    todoArr[i].index = i + 1;
+  }
+  renderTodo();
+  localStorage.setItem('TODO', JSON.stringify(todoArr));
+};
+
+const checkValue = (box) => {
+  if (box.checked) {
+    return true;
+  }
+  return false;
+};
+
+function checkTodo(todoId, box) {
+  box.nextElementSibling.classList.toggle('lineThrough');
+  todoArr[todoId].complete = checkValue(box);
+  if (todoArr[todoId].complete === true) {
+    box.complete = true;
+    box.nextElementSibling.classList.add('lineThrough');
+  }
+  localStorage.setItem('TODO', JSON.stringify(todoArr));
+}
+
+const completeTodo = () => {
+  todoArr = todoArr.filter((obj) => obj.complete !== true);
+  renderTodo();
+  for (let i = 0; i < todoArr.length; i += 1) {
+    todoArr[i].index = i + 1;
+  }
+  localStorage.setItem('TODO', JSON.stringify(todoArr));
```

Comment:
As you can see, you wrote the (localStorage,setItems) multiple times in lines 63, 80, and 89 therefore promoting repetition. I think a quick solution is to make that a function and call the function whenever its needed.